src/paginator.js

```diff
@@ -0,0 +1,300 @@
+import assign from 'lodash.assign';
+import omit from 'lodash.omit';
+
+
+export const styles = Object.freeze({
+  pageNumber: 1,
```

Comment:
> I generally find enums to be a weak code smell—they're often used for swtich statements. I view switch statements as a code smell because it hints that we're not using good OO. For example: instead of creating a single class with behavior that branches off an enum: create two classes: PageNumberPaginator and LimitOffsetPaginator (and you can see that is how it is done in drf). Now we have eliminated branching logic in our source (a big win) and the consumer will decide which class to use.

I semi-disagree with switch statements being a code-smell, but....

Following SOLID, the S (Single Responsibility) is violated by not having two different classes based on the style. We should definitely break this up into a parent and child classes.

(And doing so means `Paginator` takes only three arguments)